crates/solvers/src/domain/order.rs

```diff
@@ -0,0 +1,54 @@
+//! The domain object representing a CoW Protocol order.
+
+use crate::domain::eth;
+
+/// A CoW Protocol order in the auction.
+#[derive(Debug, Clone)]
+pub struct Order {
+    pub uid: Uid,
+    pub sell: eth::Asset,
+    pub buy: eth::Asset,
+    pub side: Side,
+    pub class: Class,
+}
+
+/// UID of an order.
+#[derive(Debug, Clone, Copy)]
+pub struct Uid(pub [u8; 56]);
+
+/// The trading side of an order.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Side {
+    /// An order with a fixed buy amount and maximum sell amount.
+    Buy,
+    /// An order with a fixed sell amount and a minimum buy amount.
+    Sell,
+}
+
+/// The order classification.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Class {
+    Market,
+    Limit,
+    Liquidity,
+}
+
+/// A user order marker type.
+#[derive(Debug)]
+pub struct UserOrder<'a>(&'a Order);
```

Comment:
This seems like it might become awkward because the type can only hold a reference. This means you can't pass user order by value. But if you make the type own an Order then the only way to create a user order is to consume the original order. Alternatively you could add a ref user order and an owning user order but that feels awkward too. I don't know how big of a problem this is or how to solve it. Also not blocking the PR.

Reply:
I used a reference for now as that is what I currently need. We can always change to a:
```rust
pub struct UserOrder<'a>(Cow<'a, Order>);
```

If we need to support both owned and borrowed orders.